ansible/library/pn_initial_ztp.py

```
@@ -580,90 +209,112 @@ def main():
+       cli.switch_setup_show(format='switch-name', show_headers=False):
+        cli.switch_setup_modify(switch_name=module.params['pn_current_switch'])
+        changed = True
 
-    # Update switch names to match host names from hosts file
-    if 'Updated' in update_switch_names(module, current_switch):
-        CHANGED_FLAG.append(True)
-
-    # Make switch setup static
+    # Configure the switch
     if module.params['pn_static_setup']:
-        output = make_switch_setup_static(module)
-        if output != '':
-            message += output
-
-    # Create/join fabric
-    if 'already in the fabric' in create_or_join_fabric(module, fabric_name,
-                                                        fabric_network):
-        message += ' %s: Already a part of fabric %s \n' % (current_switch,
-                                                            fabric_name)
-    else:
-        message += ' %s: Joined fabric %s \n' % (current_switch,
-                                                 fabric_name)
-        CHANGED_FLAG.append(True)
-
-    # Configure fabric control network to either mgmt or in-band
-    if 'Success' in configure_control_network(module, control_network):
-        message += ' %s: Configured fabric control network to %s \n' % (
-            current_switch, control_network)
-        CHANGED_FLAG.append(True)
-    else:
-        message += ' %s: Fabric is already in %s control network \n' % (
-            current_switch, control_network)
-
-    # Enable web api if flag is True
+        #TODO: Can't pass parameters like this, none will pass
+        #TODO: Verify setup to determine idempotency
+        cli.switch_setup_modify(**module.params)
+
+    fabric = cli.fabric_show(format='name',show_headers=False).split()
+    
+    # Make sure the switches are connected to the correct fabric
+    if module.params['pn_fabric_name'] not in fabric:
+        # TODO: un-join from existing fabrics and move to the correct fabric
```

I think so here we should throw an error saying **Switch is already part of a fabric, please reset the switches to use this playbook**... something like this.